pkgs/servers/web-apps/plausible/default.nix

```diff
@@ -0,0 +1,85 @@
+{ lib, stdenv, beamPackages, fetchFromGitHub, glibcLocales, cacert
+, mkYarnModules, nodejs, fetchpatch, nixosTests
+}:
+
+let
+  pname = "plausible";
+  version = "1.3.0";
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "plausible";
+    repo = "analytics";
+    rev = "v${version}";
+    sha256 = "03lm1f29gwwixnhgjish5bhi3m73qyp71ns2sczdnwnbhrw61zps";
+  };
+
+  mixDeps = beamPackages.fetchMixDeps {
+    pname = "${pname}-deps";
+    inherit src version;
+    sha256 = "sha256-66zSdYmis3UnbdLkPi649RbPbMPI5gVeFlaMekOy5CQ=";
+  };
+
+  yarnDeps = mkYarnModules {
+    pname = "${pname}-yarn-deps";
+    inherit version;
+    packageJSON = ./package.json;
+    yarnNix = ./yarn.nix;
+    yarnLock = ./yarn.lock;
+    preBuild = ''
+      mkdir -p tmp/deps
+      cp -r ${mixDeps}/phoenix tmp/deps/phoenix
+      cp -r ${mixDeps}/phoenix_html tmp/deps/phoenix_html
+    '';
+    postBuild = ''
+      echo 'module.exports = {}' > $out/node_modules/flatpickr/dist/postcss.config.js
+    '';
+  };
+in beamPackages.mixRelease {
+  inherit pname version src mixDeps;
+
+  nativeBuildInputs = [ nodejs ];
+
+  patches = [
+    # Allow socket-authentication against postgresql. Upstream PR is
+    # https://github.com/plausible/analytics/pull/1052
+    (fetchpatch {
+      url = "https://github.com/Ma27/analytics/commit/f2ee5892a6c3e1a861d69ed30cac43e05e9cd36f.patch";
+      sha256 = "sha256-JvJ7xlGw+tHtWje+jiQChVC4KTyqqdq2q+MIcOv/k1o=";
+    })
+  ];
+
+  passthru = {
+    tests = { inherit (nixosTests) plausible; };
+    updateScript = ./update.sh;
+  };
+
+  postPatch = ''
+    # Without this modification, tzdata tries to write in its store-path:
+    # https://github.com/lau/tzdata#data-directory-and-releases
+    echo 'config :tzdata, :data_dir, (System.get_env("PLAUSIBLE_TZDATA") || "/tmp/plausible_tzdata")' \
+      >> config/config.exs
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    mkdir -p $out
+    ln -sf ${yarnDeps}/node_modules assets/node_modules
+    mix deps.compile --path $out --no-deps-check
+    mix compile --no-deps-check --path $out
+
+    npm run deploy --prefix ./assets
```

Comment:
the mix digest task is missing. On some applications it's not required. Just mentioning it here in case you forgot.

Reply:
Yeah, should've added a comment in the first place, sorry!
The thing is, when I run `mix phx.digest`, it was complaining about missing dependencies and I didn't really understand why. Do you have an idea or want to take a look? Otherwise I can do it next week (IIRC this is used for minify tasks and such, I tested plausible a bit and it seemed to run OK for me, so maybe there's no actual need anyways).

Reply:
you need to run it with the `--no-deps-check` flag. Let me know if you still have a problem and I'll take a look.

Reply:
The digest task is more for cache invalidation. If you don't run it, when an end user loads the website, there is a chance that it will just use the cached version. Digest gives a unique name that will make sure that a new version is downloaded when there is a need.

Reply:
Hmm would you mind taking a look? Even with `--no-deps-check` it errors out because of dependency issues.